src/MongoDBRepositoryFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Broadway\ReadModel;
+
+use App\MongoDB\Client;
+use Broadway\ReadModel\RepositoryFactoryInterface;
+use Broadway\Serializer\SerializerInterface;
+
+/**
+ * @author Robin van der Vleuten <robin@webstronauts.co>
+ */
+class MongoDBRepositoryFactory implements RepositoryFactoryInterface
+{
+    /**
+     * @var Client
+     */
+    private $mongodb;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * Constructor.
+     *
+     * @param Client $mongodb
+     * @param SerializerInterface $serializer
+     */
+    public function __construct(Client $mongodb, SerializerInterface $serializer)
+    {
+        $this->mongodb = $mongodb;
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($name, $class)
```

Comment:
what exactly are $name and $class?

Reply:
can they be type hinted?

Reply:
They are coming from the [broadway interface](https://github.com/broadway/broadway/blob/master/src/Broadway/ReadModel/RepositoryFactory.php#L25). So they need to be type hinted there as well.